### ISSUE TITLE
Review focus management for ajax modal

### ISSUE THREAD

**Issue by thatbudakguy** [sul-dlss/SearchWorks#3518]
It looks like we have a modal that we pre-render and use to fill with information from ajax requests:

https://github.com/sul-dlss/SearchWorks/blob/82e4a15ba35e156825d4749c5314d310804bab79/app/views/shared/_ajax_modal.html.erb#L1-L6

SiteImprove points out that it shouldn't be hidden with `aria-hidden` if it could have keyboard focus set to it (because of its `tabindex`).

Modals generally require some extra work in terms of focus management — the modal should be visually hidden and not keyboard-focusable until someone triggers it, at which point you immediately move focus into the modal and prevent focus from leaving the modal (intentional focus trap). When it's closed, you're supposed to move focus back to wherever on the page the modal was triggered from. Usually this happen in the JavaScript; I'm not sure what bootstrap's does, but we have some that the commit history suggests is Blacklight behavior that we copied over, so maybe modern Blacklight doesn't have this issue?

We probably have a bit of due diligence to confirm how correctly we're implementing this pattern. At a minimum, we can remove the `aria-hidden` to clear the alert in SiteImprove. There's also now apparently an `aria-modal=true` attribute which helps with focus management in this scenario — see [the w3c modal examples](https://www.w3.org/WAI/ARIA/apg/patterns/dialog-modal/examples/dialog/) for more info.

The "focusable aria-hidden element" is apparently a level A violation. [See it in SiteImprove](https://my2.siteimprove.com/Accessibility/1059585/NextGen/Issue/1?ruleId=17&pageSegments=&issueKind=1&tagFilterType=2&conformance=&siteTargetIssueKinds=1,2&wcagVersion=21&siteTarget.wcagVersion=21&siteTarget.conformanceLevels=0,1,3,4&siteTarget.issueKinds=1,2).

**Comment by jcoyne** [comment]
@thatbudakguy this is a bootstrap managed modal. The bootstrap adds `aria-hidden="true"` when the modal is minimized. https://github.com/twbs/bootstrap/blob/main/js/src/modal.js#L245

Bootstrap does do focus trapping within modals (but maybe not the old version we are using).  

Versions of bootstrap since 4.2 also utilizes `aria-modal=true`

**Comment by thatbudakguy** [comment]
blocked by bs4 upgrade

**Comment by thatbudakguy** [comment]
Resolved by moving to bootstrap 4.
